src/main/java/api/v1/endpoints/SchoolsEndpoint.java

```diff
@@ -24,12 +24,20 @@ public OpenApiDocumentation configureDocs(OpenApiDocumentation docs) {
           openApiOperation.description(
               "This endpoint returns an object where keys are school codes, and values are their full names.");
           openApiOperation.summary("Schools Endpoint");
+        })
+        .queryParam("query", String.class, openApiParam -> {
```

Comment:
I think it's fine if we add this, but I'm not sure what the use case would be; it sends a little less information, yes, but ultimately we're talking on the scale of 23ms to go through all the schools and return the entire list, and it's returned as a map so the client can just filter it using a lookup if need be.

Reply:
Yeah, I'm supposed. I was just lazy when I was working on frontend and want to specifically finding names of the code itself

Reply:
lmao front end must be really rough if it's easier to rewrite the endpoint